src/OrleansServiceBus/Providers/Streams/EventHub/IEventHubSettings.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+namespace OrleansServiceBusUtils.Providers.Streams.EventHub
+{
+    public interface IEventHubSettings
```

Comment:
This pattern was selected to allow application developers to provide the servicebus configuration through their own implementation of this interface.  The reason this is important is that the servicebus connection string is security sensitive information so storing it in cluster configuration xml will be undesirable in many cases.
Developers can avoid this by accessing their secure configuration and programmatically setting up the provider, but that requires custom initialization logic and translating from their preferred (secure) configuration system to Orleans provider configuration.
The approach used here allows them to inject there own configuration object (reading from secure configuration) via the service locator.  This is a very convenient method for service developers to provide secure configuration to the provider from their preferred configuration system.